src/core/communication.cpp

```diff
@@ -123,12 +123,13 @@ void init(std::shared_ptr<boost::mpi::environment> mpi_env) {
 }
 
 void deinit() {
-  ErrorHandling::deinit_error_handling();
-  Communication::m_callbacks.reset();
-
```

Comment:
For just curiosity, are there any problems with the previous order: Communication::m_callbacks.reset(); kokkos_handle.reset();?